api/src/main/java/jakarta/mvc/form/FormMethodOverwriter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package jakarta.mvc.form;
+
+import jakarta.ws.rs.container.ContainerRequestContext;
+/**
+ * <p>Form method overwriter</p> are used to overwrite a HTML form's HTTP method to be able to use
+ * other verbs such as {@code PATCH} or {@code DELETE}. They are discovered using Jakarta Contexts and Dependency Injection.
+ * </p>
+ * 
+ * @author Tobias Erdle
+ * @since 2.1
+ */
+public interface FormMethodOverwriter {
+    
+    /**
+     * Property that can be used to enable the Form method overwrite mechanism for an application.
+     * Values of this property must be of type {@link FormMethodOverwriter.Options}.
+     */
+    String FORM_METHOD_OVERWRITE_STATUS = "jakarta.mvc.form.FormMethodOverwrite";
+
+    /**
+     * The name of the hidden form input to get the targeted HTTP method.
+     */
+    String HIDDEN_FIELD_NAME = "_method";
```

Comment:
Two comments:

1. Does anyone know if there is some kind of "de-facto standard" for the name of this parameter? AFAIK other frameworks support this mechanism as well and if there is some kind of standard name, it would be cool to use it.
2. Maybe we should allow to customize the name of the form field via some configuration parameter?

Reply:
All frameworks I know use `_method` (especially Spring), so this name should be the `de-facto` standard. I added a possibility to overwrite this name in my new draft.